packages/viewer/src/interactions/mouseInteractionHandler.ts

```diff
@@ -31,18 +37,22 @@ export class MouseInteractionHandler implements InteractionHandler {
   private primaryInteractionTypeChange = new EventDispatcher<void>();
 
   public constructor(
+    private usePointerEvents: boolean,
```

Comment:
feels to me like this would be more readable as a property on `InteractionHandler`, or a new class for interaction handlers that support whatever the common theme of interactions are here.

Reply:
Sounds good!

Reply:
+1. Instead of using conditionals, another option is to have a base class that has the interaction implementation, and delegates to subclasses for registering events. For example:

```ts
abstract class BaseInteractionHandler implements InteractionHandler {
  protected abstract registerEvents(element: HTMLElement): void;
  protected abstract deregisterEvent(): void;

  public initialize(...): void {
    this.registerEvents(element);
  }
}

class PointerInteractionHandler extends BaseInteractionHandler {
  protected registerEvents(element: HTMLElement): void {
    element.addEventListener('pointerdown', this.handleDownEvent);
  }
}
```